public/app/plugins/datasource/grafana-pyroscope-datasource/QueryEditor/PyroscopeAppIntegration.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {useEffect, useMemo, useState} from 'react';
+
+import { QueryEditorProps, TimeRange, DateTime } from '@grafana/data';
+import { getBackendSrv } from '@grafana/runtime';
+import {LinkButton} from '@grafana/ui'
+
+import { PyroscopeDataSource } from '../datasource';
+import { PyroscopeDataSourceOptions, Query } from '../types';
+
+const PYROSCOPE_APP_ID = 'grafana-pyroscope-app'
+
+/* Global promise to fetch the pyroscope app settings */
+let pyroscopeAppSettings: Promise<PyroscopeAppSettings> | null= null;
+/* Global promises to fetch pyroscope datasource settings by uid as encountered */
+const pyroscopeDatasourceSettingsByUid: Record<string, Promise<PyroscopeDatasourceSettings>> = {};
+
+export type Props = QueryEditorProps<PyroscopeDataSource, Query, PyroscopeDataSourceOptions>;
+
+/** A subset of the app settings that are relevant for this integration */
+type PyroscopeAppSettings = {
+  enabled: boolean;
+  jsonData: {
+    backendUrl: string;
+    basicAuthUser: string;
+  }
+}
+
+/** A subset of the datasource settings that are relevant for this integration */
+type PyroscopeDatasourceSettings = {
+  url: string;
+  basicAuthUser: string | number; // This might be configured as a number, so we will want to ensure compare for string-equivalence
+}
+
+export function PyroscopeAppIntegration(props: Props) {
+
+  const [appPlugin, setAppPlugin] = useState<PyroscopeAppSettings>()
+  const [datasource, setDatasource] = useState<PyroscopeDatasourceSettings>()
+
+  const {datasource:{uid: datasourceUid}, query, range} = props;
+
+  useEffect(()=>{
+
+    if (pyroscopeAppSettings == null) {
+      pyroscopeAppSettings = getBackendSrv().get<PyroscopeAppSettings>(`/api/plugins/${PYROSCOPE_APP_ID}/settings`)
+    }
+
+    pyroscopeAppSettings.then(setAppPlugin);
+    
+  }, [])
+
+  useEffect(()=>{
+    let datasourceSettings = pyroscopeDatasourceSettingsByUid[datasourceUid];
+
+    if (datasourceSettings == null) {
+      datasourceSettings = getBackendSrv().get<PyroscopeDatasourceSettings>(`/api/datasources/uid/${datasourceUid}`)
+      pyroscopeDatasourceSettingsByUid[datasourceUid] = datasourceSettings;
```

Comment:
The data source settings should be passed in the `QueryEditorProps` so I don't think we need to access it like this.

Reply:
I will check again, but the version of the datasource settings I saw there didn't include the auth information.

I suspect in the case of QueryEditorProps, it's coming from the `/api/datasources` endpoint which strips them, versus the per-uid endpoint which is used by the datasource config screen.

Reply:
ah ok we need a basicAuth user. Hmm tbh not sure if this is great I mean I feel like this is a bit security leak because if user does not have admin access to the data source they should not be able to get to the auth user.